api/python/t4/packages.py

```diff
@@ -857,9 +864,13 @@ def set(self, logical_key, entry=None, meta=None):
                     url = make_s3_url(bucket, key, version)
             entry = PackageEntry([url], size, None, orig_meta)
         elif isinstance(entry, PackageEntry):
+            validate_key(_to_singleton(entry.physical_keys))
```

Comment:
Same here.

Reply:
Note that since we regard setting an entry as a `PackageEntry` as a special case, I thought about leaving this check out, but I couldn't because `set_dir` (surprisingly, in my opinion) uses this code path.